sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/streaming/WriteToStream.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.streaming
+
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.connector.catalog.Table
+import org.apache.spark.sql.streaming.OutputMode
+
+/**
+ * Used to create a [[StreamExecution]].
+ */
+case class WriteToStream(
+    name: String,
+    checkpointLocation: String,
+    sink: Table,
+    outputMode: OutputMode,
+    deleteCheckpointOnStop: Boolean,
+    inputQuery: LogicalPlan) extends LogicalPlan {
+
+  override def isStreaming: Boolean = true
+
+  override def output: Seq[Attribute] = Nil
+
+  override def children: Seq[LogicalPlan] = Nil
```

Comment:
let's add a comment to explain why `inputQuery` is not a child: it's already a resolved plan from a DataFrame.

Reply:
The same to `WriteToStreamStatement`

Reply:
BTW `DataFrameWriter` creates `ReplaceTableAsSelectStatement`, `AppendData`, etc. with a resolve input query plan, and repeated analysis may not be an issue at all.

Reply:
For streaming queries, query compilation overhead matters even less.

Reply:
Why? Is it because streaming nodes are usually skipped?

Reply:
because streaming queries are long-running, and `WriteToStream` is resolved only once.

Reply:
I see. Thanks!

Reply:
Thanks, done in 04e1b23

Reply:
@xuanyuanking how do you think about just allowing repeated analysis?

Reply:
Sure. Let me try.

Reply:
Done in 7ec1afb